app/controllers/declaration/SealController.scala

```diff
@@ -153,13 +153,16 @@ class SealController @Inject()(
           Future.successful(navigator.continueTo(routes.SealController.displaySealSummary(mode, cachedContainer.id)))
     )
 
-  private def updateCache(container: Container)(implicit req: JourneyRequest[AnyContent]) =
+  private def updateCache(updatedContainer: Container)(implicit req: JourneyRequest[AnyContent]) =
     updateExportsDeclarationSyncDirect(model => {
 
       val updatedContainers =
         model.containerData
           .map(_.containers)
-          .fold(Seq(container))(_.filterNot(_.id == container.id) :+ container)
+          .fold(Seq(updatedContainer))(_.map {
```

Comment:
Please move map logic for Containers object

Reply:
Good idea.

Reply:
Can you take a 2nd look @bambuchaAdm